src/core/thread/network_data_local.cpp

```diff
@@ -69,6 +68,44 @@ Error Local::AddOnMeshPrefix(const OnMeshPrefixConfig &aConfig)
     // of an IEEE 802.15.4 interface MUST have a length of 64 bits.
     VerifyOrExit(!aConfig.mSlaac || aConfig.mPrefix.mLength == OT_IP6_PREFIX_BITSIZE, error = kErrorInvalidArgs);
 
+    // Validate Prefix
+    VerifyOrExit((aConfig.GetPrefix().GetLength() > 0) && aConfig.GetPrefix().IsValid(), error = kErrorInvalidArgs);
+
+    // Validate Preference
+    switch (aConfig.mPreference)
+    {
+        case OT_ROUTE_PREFERENCE_LOW:
+        case OT_ROUTE_PREFERENCE_MED:
+        case OT_ROUTE_PREFERENCE_HIGH:
+            break;
+        default:
+            ExitNow(error = kErrorInvalidArgs);
+    }
+
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
+    if (aConfig.mDp)
+    {
+        // Check if domain prefix is already configured
+        OnMeshPrefixConfig prefixConfig;
+        VerifyOrExit(Get<BackboneRouter::Local>().GetDomainPrefix(prefixConfig) == kErrorNotFound, error = kErrorAlready);
+    }
+    else
+#endif
+    {
+        // Check if address already configured
+        VerifyOrExit(!aConfig.GetPrefix().ContainsPrefix(Get<Mle::MleRouter>().GetMeshLocalPrefix()), error = kErrorAlready);
+        // Verify network data contains enough space
+        VerifyOrExit(CheckPrefixBufferSize(aConfig.GetPrefix()) == kErrorNone, error = kErrorInvalidArgs);
+    }
+
+exit:
+    return error;
+}
+
+Error Local::AddOnMeshPrefix(const OnMeshPrefixConfig &aConfig)
```

Comment:
Since `AddOnMeshPrefix` is also called by other modules besides `otBorderRouterAddOnMeshPrefix`, we should call `ValidateOnMeshPrefix` in `AddOnMeshPrefix` again to make sure the prefix is valid?

Reply:
@simonlingoogle right. I have added external validation.

Reply:
There are calls to `AddOnMeshPrefix` that bypasses `ValidateOnMeshPrefix` (e.g. in routing_manager.cpp).

Should we just call `ValidateOnMeshPrefix` again in `AddOnMeshPrefix`? `otBorderRouterAddOnMeshPrefix` would then call `ValidateOnMeshPrefix` twice but that does not seems to be a real issue anyway.